compiler/src/typed/typecore.re

```diff
@@ -977,18 +977,19 @@ and type_expect_ =
         Builtin_types.type_box @@
         newvar(~name="a", ()),
       );
-    let val_ = type_expect(env, sval, ty_expected_explained);
+    let val_ = type_expect(env, sval, mk_expected(newvar(~name="a", ())));
```

Comment:
What is with `~name="a"`? I assume it is just constructing a dummy var, but does `a` ever show up in errors or what if it is shadowed or something?

Reply:
Ah yeah, there's no reason to provide a name here. It wouldn't ever show up in an error by virtue of everything will typecheck with it. As for shadowing, that's really only a concern during parsing. After typechecking, all identifiers are unique, even if they have the same name (since every identifier has its own id).

Reply:
```suggestion
    let val_ = type_expect(env, sval, mk_expected(newvar()));
```